### PR TITLE
Add a feature to emit task-received and task-started events for batch…

### PR DESCRIPTION
**Pull request by srinivaas14**
…ed requests

- The above mentioned events will be useful to track the state of each request (which is a part of batched request list)
  in the monitoring tools like Flower

issue ref: https://github.com/clokep/celery-batches/issues/33

### CODE REVIEW

**Comment by clokep** [comment]
Thanks! Hopefully I'll have time to look at this soon. 👍

**Comment by srinivaas14** [comment]
> Thanks! Hopefully I'll have time to look at this soon.

Great ! Looking forward to contribute more features. 👍

File: celery_batches/__init__.py [hunk 1]
```diff
@@ -304,6 +322,11 @@ def flush(self, requests):
 
         def on_accepted(pid, time_accepted):
             [req.acknowledge() for req in acks_late[False]]
+            for req in requests:
+                # Start time of the task, which will be useful in calculating runtime of the task
+                req.start_time = time_accepted
```
Comment (clokep) [inline celery_batches/__init__.py:327]: I think Celery calls this `time_start`?

https://github.com/celery/celery/blob/8d6778810c5153c9e4667eed618de2d0bf72663e/celery/worker/request.py#L69

**Review by clokep — COMMENTED** [review]
Sorry for the extreme delay on this!

Overall this looks great, for my own notes it is pretty much copying bits of the main `Request` class and some of the `default_strategy`. I left one comment -- I think this is a typo, but I'm not sure?

**Comment by clokep** [comment]
Oh, any idea on if we can write tests for this? I imagine it wouldn't be too bad, just enabling some signals and ensuring we receive them?

**Comment by kinoute** [comment]
Any news? I'm interested in tracking each task (part of a batch) I'm receiving. Right now it's kind of confusing because while the buffer gets populated and before it gets triggered, we have zero logs. In celery "classic", we could see each task received which was great.

**Comment by clokep** [comment]
> Any news? I'm interested in tracking each task (part of a batch) I'm receiving.

Looks like the author isn't interested in finishing this up, unfortunately. I'm going to close this since it bitrotted.

> Right now it's kind of confusing because while the buffer gets populated and before it gets triggered, we have zero logs. In celery "classic", we could see each task received which was great.

I haven't focused much on flower compatibility in the past (since that's not a thing I use), but would be open to contracting to add support for it. (Although I'm unsure if you're looking for events or just want e.g. log lines.)

**Pull request closed without merging**